lib/ansible/plugins/strategy/__init__.py

```diff
@@ -558,7 +558,7 @@ def search_handlers_by_notification(self, notification: str, iterator: PlayItera
                     handler.fattributes.get('listen'),
                     listeners,
                     templar,
-                )
+                ).copy()
                 if handler._role is not None:
                     for listener in listeners.copy():
```

Comment:
This second copy probably isn't needed anymore

Reply:
It is still needed as `listeners` are mutated within the `for` loop.